app/src/main/res/layout-land/fragment_main.xml

```diff
@@ -0,0 +1,75 @@
+<?xml version="1.0" encoding="utf-8"?>
```

Comment:
Is it necessary to create MainFragment for landscape mode?
I mean can't we make changes to the existing layout to make it compatible with landscape mode?

Reply:
@rob729 It's necessary because the action bar title is not adjusting as we have hardcoded the margin for the portrait mode.
It could have auto adjusted if it wasn't hardcoded and gravity was set as center in the portrait mode itself (though I m not fully sure about it)

Reply:
Have you tried using a single file to achieve the desired layout for both portrait and landscape mode?

Reply:
@rob729 Yes I have. But the same problem arises since we are manually setting action bar title. It's not possible using one layout file in this scenario.

Reply:
Using center_horizontal gravity can't we achieve the desired layout in both modes?

Reply:
yes, it can be done. But you asked to hardcode the margin value in portrait mode as it was coming little bit left when aligned using "center_horizontal gravity"

Reply:
You can use both at the same time. Use center_horizontal gravity and marginStart value to achieve the desired output. A little bit offset is okay.

Reply:
So shall I make these changes? Any other changes reuired?

Reply:
What about the NewProfileFragment landscape layout? Is it really required if we add scrollview in the portrait layout itself?

Reply:
@rob729 I can try, but what's the use of scroll view in portrait layout? Just to avoid landscape file we can, other than that it has no use.

Reply:
According to you which is a better approach, copying almost the same code in another layout file or just adding scrollview in the existing layout file?

Reply:
Both methods has their own problem, I'll go with what you say. I'll remove the layout file. Anything else that needs to be improved?

Reply:
No rest looks fine.